### PR TITLE
bib: bump go to v1.21 and images lib to v0.69.0

### PR DESCRIPTION
**Pull request by yoheiueda**
This patch bumps the version of the images library to v0.69.0 as well as go to v1.21, since the library requires the Go version.

This version includes a necessary fix to support partition layouts for non-UEFI platforms. https://github.com/osbuild/images/pull/758

I created this patch using the following commands.
```
go mod edit -go 1.21 -require github.com/osbuild/images@v0.69.0
go mod tidy
```

### CODE REVIEW

**Comment by yoheiueda** [comment]
images v0.69.0 causes the following lint error.

https://github.com/osbuild/bootc-image-builder/actions/runs/9757284982/job/26997810315?pr=515#step:6:31
```
Running [/home/runner/golangci-lint-1.55.2-linux-amd64/golangci-lint run --path-prefix=bib --timeout 5m0s] in [/home/runner/work/bootc-image-builder/bootc-image-builder/bib] ...
  Error: bib/cmd/bootc-image-builder/partition_tables.go:4:2: "github.com/osbuild/images/pkg/arch" imported and not used (typecheck)
  	"github.com/osbuild/images/pkg/arch"
  	^
  
  Error: issues found
```

This error was due to the lack of build tags specified to `golangci-lint`

https://github.com/osbuild/bootc-image-builder/blob/f5e218ae5e3a985c6a0891dc3959a2056136db59/build.sh#L7

File: .golangci.yaml
Comment (yoheiueda) [inline .golangci.yaml:1]: This is a configuration file for `golangci-lint` in this CI test.

https://github.com/osbuild/bootc-image-builder/blob/f5e218ae5e3a985c6a0891dc3959a2056136db59/.github/workflows/tests.yml#L41-L46

File: .golangci.yaml
Comment (yoheiueda) [inline .golangci.yaml:6]: This is a configuration file for `golangci-lint` in this CI test.

https://github.com/osbuild/bootc-image-builder/blob/f5e218ae5e3a985c6a0891dc3959a2056136db59/.github/workflows/tests.yml#L41-L46

**Review by achilleas-k — CHANGES_REQUESTED** [review]
You need to also update the go version in the `Lint & unittests` job, otherwise the runner will use go 1.20 to lint the project.  This always results in weird errors like the ones we're seeing (unused imports).

The build flags added in the second commit shouldn't be necessary.

**Comment by achilleas-k** [comment]
I updated the Go version in the GitHub workflow.

**Review by mvo5 — APPROVED** [review]
Thank you!

**Comment by achilleas-k** [comment]
The konflux job is stuck waiting for approval but I don't see any way to approve it -_-

**Review by achilleas-k — APPROVED** [review]

**Comment by achilleas-k** [comment]
Never mind.  The PR was just blocked on my previews change request.

**Pull request merged**
